src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs

```diff
@@ -680,6 +680,29 @@ public QueryAddress AddressBalance(string address)
          };
       }
 
+
+      /// <summary>
+      /// Calculates the balance for specified list of addresses.
+      /// </summary>
+      /// <param name="addresses"></param>
+      public List<AddressBalanceResult> MultipleAddressBalances(List<string> addresses)
+      {
+         var result = new ConcurrentBag<AddressBalanceResult>();
+
+         Parallel.ForEach(addresses, address =>
```

Comment:
I wonder if parallel is really needed? some of this calls can take time if the address has many entries in history and has not been computed before